src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -0,0 +1,83 @@
+package com.tstepnik.planner.service;
+
+import com.tstepnik.planner.domain.Importance;
+import com.tstepnik.planner.domain.Task;
+import com.tstepnik.planner.domain.User;
+import com.tstepnik.planner.exceptions.TaskNotFoundException;
+import com.tstepnik.planner.repository.TaskRepository;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TaskService {
+
+    private final Importance DEFAULT_IMPORTANCE = Importance.MINOR;
+
+    private final TaskRepository taskRepository;
+    private final AuthService authService;
+//TODO here add whole logic with if and exception and use loggedUser here!
+
+
+    public TaskService(TaskRepository taskRepository, AuthService authService) {
+        this.taskRepository = taskRepository;
+        this.authService = authService;
+    }
+
+    public List<Task> findAll() {
+        return taskRepository.findAll();
+    }
+
+    public Optional<Task> findTaskById(Long id) {
+        return taskRepository.findById(id);
+    }
+
+    public List<Task> getUserTasks() {
+        User user = authService.getLoggedUser();
+        return taskRepository.findAllByUserId(user.getId());
+    }
+
+    public Task addTask(Task task) {
+        User user = authService.getLoggedUser();
+        if (task.getImportance() == null) {
+            task.setImportance(DEFAULT_IMPORTANCE);
+        }
+        task.setUserId(user.getId());
+        return taskRepository.save(task);
+    }
+
+    public Task updateTask(Task task, Long taskId) {
+        User user = authService.getLoggedUser();
+        Optional<Task> checkedTask = findTaskById(taskId);
+        if (checkedTask.isEmpty()) {
+            return taskRepository.save(task);
```

Comment:
This method is update, not upsert. So if there is no entity with given email we should throw exception. You can also read something about upsert if you want to.

Reply:
Ok I add method with "get" to task repository so if there is no task spring automatically will throws exception.  How about that solution ?

Reply:
You dont need to add getById to repository because it is inherited from JpaRepository I think.